scrapy/http/request/__init__.py

```diff
@@ -50,8 +50,8 @@ def _set_url(self, url):
         if not isinstance(url, six.string_types):
             raise TypeError('Request url must be str or unicode, got %s:' % type(url).__name__)
 
-        url = to_native_str(url, self.encoding)
-        self._url = escape_ajax(safe_url_string(url))
+        s = safe_url_string(url, self.encoding)
```

Comment:
Does it mean users now must use `yield scrapy.Request(url, ..., encoding=response.encoding)` in order to get correct URL escaping?